app/api/delete_indices_endpoint.py

```diff
@@ -0,0 +1,36 @@
+from fastapi import APIRouter, Depends, HTTPException
+from app.config import server
+from tracardi.config import NAME
+from tracardi.service.storage.elastic_client import ElasticClient
+from elasticsearch import ElasticsearchException
+from .auth.permissions import Permissions
+from typing import Optional
+from hashlib import md5
+
+router = APIRouter(
+    dependencies=[Depends(Permissions(roles=["admin"]))]
+)
+
+
+@router.delete("/indices/version/{version}", tags=["index"], include_in_schema=server.expose_gui_api)
+async def delete_old_indices(version: str, codename: Optional[str] = None):
+
+    if codename is None:
+        codename = md5(version.encode('utf-8')).hexdigest()[:5]
+
+    if codename == NAME:
```

Comment:
Code name is not enough. Code names can be the same but versions not. Comparing two version is in the version class (I guess). Create an object out of version, codename and use objects.